pkg/piecestore/psserver/agreementsender/agreementsender.go

```diff
@@ -10,121 +10,96 @@ import (
 	"github.com/zeebo/errs"
 	"go.uber.org/zap"
 	"golang.org/x/net/context"
-	"google.golang.org/grpc"
 
-	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/kademlia"
 	"storj.io/storj/pkg/pb"
 	"storj.io/storj/pkg/piecestore/psserver/psdb"
 	"storj.io/storj/pkg/provider"
 	"storj.io/storj/pkg/storj"
-	"storj.io/storj/pkg/utils"
+	"storj.io/storj/pkg/transport"
 )
 
 var (
-	defaultCheckInterval = flag.Duration("piecestore.agreementsender.check-interval", time.Hour, "number of seconds to sleep between agreement checks")
-	defaultOverlayAddr   = flag.String("piecestore.agreementsender.overlay-addr", "127.0.0.1:7777", "Overlay Address")
-
+	//todo: cache kad responses if this interval is very small
+	defaultCheckInterval = flag.Duration("piecestore.agreementsender.check-interval", time.Hour, "duration to sleep between agreement checks")
 	// ASError wraps errors returned from agreementsender package
 	ASError = errs.Class("agreement sender error")
 )
 
 // AgreementSender maintains variables required for reading bandwidth agreements from a DB and sending them to a Payers
 type AgreementSender struct {
-	DB       *psdb.DB
-	overlay  overlay.Client
-	identity *provider.FullIdentity
-	errs     []error
+	DB        *psdb.DB
+	log       *zap.Logger
+	transport transport.Client
+	kad       *kademlia.Kademlia
 }
 
-// Initialize the Agreement Sender
-func Initialize(DB *psdb.DB, identity *provider.FullIdentity) (*AgreementSender, error) {
-	overlay, err := overlay.NewClient(identity, *defaultOverlayAddr)
-	if err != nil {
-		return nil, err
-	}
-
-	return &AgreementSender{DB: DB, identity: identity, overlay: overlay}, nil
+// New creates an Agreement Sender
+func New(log *zap.Logger, DB *psdb.DB, identity *provider.FullIdentity, kad *kademlia.Kademlia) *AgreementSender {
```

Comment:
It should be possible to use a general interface `dht.DHT` instead of Kademlia directly.

Reply:
`dht.DHT` will be removed to simplify the code.

Reply:
oh, I saw it in recent JT code and though its current design :) thanks for info